### ISSUE TITLE
[Bug]: Search bar returns results even when "Enable client side search" is disabled

### ISSUE THREAD

**Issue by shadabbuchh** [appsmithorg/appsmith#15386]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

In Table v2 when we disable the `Enable client side search` property and then enter some text in the Search bar, the search returns results.

https://www.loom.com/share/9c5527a7a6b6426d8d999a7ee697a1c4

### Steps To Reproduce

1. D&D a Table widget.
2. Bind it with an API/Query.
3. Disable the `Enable client side search` property.
4. Now enter any text in the Search bar.

**Expected:**  The search should not filter out data.
**Actual:** Upon entering any text in the search bar, it returns results based on the search.

### Version

Cloud

**Comment by sbalaji1192** [comment]
@dilippitchika @shadabbuchh This is a known behavior in Table v1 as well. When you disable `enable client side search`  and don't have any binding in `onSearchTextChanged` property, we still go ahead and do client-side search.
should we keep doing that or strictly follow the `enable client side search` property?

**Comment by dilippitchika** [comment]
No, we should not search on client side, when this switch is turned off. This will be a bigger change. Keeping this open.

**Comment by Naveen-Goud** [comment]
Hii @carinanfonseca @shadabbuchh trying to reproduce this issue.
